web/client/utils/CoordinatesUtils.js

```diff
@@ -844,6 +844,28 @@ const CoordinatesUtils = {
             crs: 'EPSG:4326'
         };
     },
+    calculateCircleRadiusFromPixel: (coordinatesFromPixelConverter, projection, pixel = {}, latlng = {}, pixelRadius, defaultRadius = 0.01) => {
+        const {lat, lng} = latlng;
+
+        if (isNumber(lat) && !isNaN(lat) &&
+            isNumber(lng) && !isNaN(lng) &&
+            isNumber(pixel.x) && !isNaN(pixel.x) &&
+            isNumber(pixel.y) && !isNaN(pixel.y)) {
+            const radiusA = [lng, lat];
+            const pixelCoords = isFunction(coordinatesFromPixelConverter) ? coordinatesFromPixelConverter([
+                pixel.x,
+                pixel.y >= pixelRadius ? pixel.y - pixelRadius : pixel.y + pixelRadius
+            ]) : null;
+            const radiusB = pixelCoords &&
```

Comment:
You should draw the circle in the current map projection, so it remains a circle and the query looks similar (and you have also a chance that the server supports the CRS you're using). In this case the point is near to both the states 
![image](https://user-images.githubusercontent.com/1279510/81544649-166fb680-9378-11ea-90e6-be9249b936f2.png)
But the query will be something like this
![image](https://user-images.githubusercontent.com/1279510/81544756-3901cf80-9378-11ea-9a43-1b926d9ad73a.png)
So this is the result, that is wrong. 
![image](https://user-images.githubusercontent.com/1279510/81544853-53d44400-9378-11ea-8a18-0c6d627b0b22.png)

Convert instead the latLong in the current map projection and draw the circle in the projected space